File: infra-job/src/main/java/io/infra/structure/job/properties/JobProperties.java
package io.infra.structure.job.properties;

import lombok.Data;
import io.infra.structure.job.constants.Const;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

/**
 * @author qinglinl
 * Created on 2022/3/8 4:00 下午
 */
@Data
@Configuration
@ConfigurationProperties(prefix = Const.configPrefix)
public class JobProperties {
    /**
     * 是否启用
     */
    private boolean enabled;
    /**
     * job admin地址
     */
    private String adminAddresses;
    /**
     * 鉴权Token
     */
    private String accessToken;
    /**
     * app名称
     */
    private String appName;
    /**
     * 地址
     */
    private String address;
    /**
     * ip
     */
    private String ip;
    /**
     * 端口
     */
    private int port = 9999;
    /**
     * log地址
     */
    private String logPath = System.getProperty("user.home") + "/logs/task-center/jobhandler";
    /**
     * 日志保留天数
     */
    private int logRetentionDays = 2;
}

File: infra-job/src/main/kotlin/io/infra/structure/job/properties/JobProperties.kt
package io.infra.structure.job.properties

import io.infra.structure.job.constants.configPrefix
import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * @author liuqinglin
 * Date: 2025/5/13 18:19
 */
@ConfigurationProperties(prefix = configPrefix)
data class JobProperties(
    /**
     * 是否启用模块
     */
    var enabled: Boolean = false,

    /**
     * 注册应用名称
     */
    var appName: String? = null,

    /**
     * 注册admin地址
     */
    var adminAddresses: String? = null,

    /**
     * Token
     */
    var accessToken: String? = null,

    /**
     * 注册本机地址
     */
    var address: String? = null,

    /**
     * 本地ip
     */
    var ip :String? = null,

    /**
     * 端口号
     */
    var port: Int = 9999,

    /**
     * 日志打印目录
     */
    var logPath: String? = System.getProperty("user.home") + "/logs/task-center/jobhandler",

    /**
     * 保留日志天数
     */
    var logRetentionDays: Int = 2,
)